### PR TITLE
Azure: Added the ip_unattacheed policy

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description
1. Added the azure ip_unattached policy.
2. Changed the AWS ip_unattched to the new approach.
3. Added the unit tests for both aws and azure

### CODE REVIEW

**Review by ebattat — COMMENTED** [review]
@athiruma, did you verify that its not related to any live cluster before deletion

File: cloud_governance/policy/aws/ip_unattached.py [hunk 1]
```diff
@@ -1,52 +1,55 @@
...
+        unattached_addresses = []
         for address in addresses:
-            ip_no_used = False
             tags = address.get('Tags', [])
-            if not self._check_cluster_tag(tags=tags) or self._get_policy_value(tags=tags) not in ('NOTDELETE', 'SKIP'):
+            cleanup_result = False
+            ip_not_used = False
+            resource_id = address.get('AllocationId')
+            cluster_tag = self._get_cluster_tag(tags=address.get('Tags'))
+            if cluster_tag not in active_cluster_ids:
```
Comment (athiruma) [inline cloud_governance/policy/aws/ip_unattached.py:30]: Check cluster tag not in active clusters

File: cloud_governance/policy/azure/cleanup/ip_unattached.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            """
+            unattached_ips = []
+            active_cluster_ids = self._get_active_cluster_ids()
+            public_ipv4_address = self.network_operations.get_public_ipv4_addresses()
+            for ip_address in public_ipv4_address:
+                tags = ip_address.get('tags')
+                cleanup_result = False
+                is_disassociated_ipv4 = False
+                cluster_tag = self._get_cluster_tag(tags=tags)
+                if cluster_tag not in active_cluster_ids:
```
Comment (athiruma) [inline cloud_governance/policy/azure/cleanup/ip_unattached.py:48]: here

**Comment by athiruma** [comment]
> @athiruma, did you verify that its not related to any live cluster before deletion ?

yes, before doing operation checking the cluster tag in active clusters

File: cloud_governance/policy/azure/cleanup/ip_unattached.py (hunk 2 above)
Comment (ebattat) [reply cloud_governance/policy/azure/cleanup/ip_unattached.py:48]: Did we add the same for AWS policy ?

Comment (athiruma) [reply cloud_governance/policy/azure/cleanup/ip_unattached.py:48]: yes

**Comment by ebattat** [comment]
I will approved it but please add todo for adding integration tests in the future

**Comment by athiruma** [comment]
Open [Issue](https://github.com/redhat-performance/cloud-governance/issues/729) for adding the integration tests.

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**